optimum/habana/transformers/generation/utils.py

```diff
@@ -193,24 +193,26 @@ def _prepare_decoder_input_ids_for_generation(
         decoder_start_token_id = self._get_decoder_start_token_id(decoder_start_token_id, bos_token_id)
         if device is None:
             device = self.device
-        if token_idx is None:
```

Comment:
@malkomes : I am not sure or cant remember why this check was there. I see you have removed it, can you comment why this is removed?

Reply:
sure @ssarkar2 , here I just re-organize the logic and it is kinda of subtle. 

Before it was something like:

 - if not static shape:
   - is batch?
     - validate batch
     - re-org batch shape for `decoder_input_ids_start` 
   - (else) not a batch
     - `decoder_input_ids_start` <- ones * `decoder_start_token_id`
 - (else) using static shape:
   - max_length 
   - `decoder_input_ids_start` <- ones (with max_length) * `decoder_start_token_id`

multiple values for `decoder_start_token_id` wouldn't work when using static shape.

now the code reads something like:

- is batch?
   - validate batch
   - re-org batch shape for `decoder_input_ids_start` 
- (else) not a batch
    - `decoder_input_ids_start` <- ones * `decoder_start_token_id`
- if is static shape:
   - max_length
   - (the `decoder_input_ids_start` now has the expected format for both cases above, so we just need to apply the "static shape" logic)
   - `decoder_input_ids_start` <- ones (with max_length) * `decoder_input_ids_start`

Let me know if that makes sense.